### PR TITLE
ip addr show pattern change

### PR DESCRIPTION
**Pull request by LiliDeng**

### CODE REVIEW

File: lisa/tools/ip.py [hunk 1]
```diff
@@ -55,9 +58,9 @@ class Ip(Tool):
     """
     __ip_addr_show_regex = re.compile(
         (
-            r"\d+: (?P<name>\w+): \<.+\> .+\n\s+link\/(?:ether|infiniband) "
-            r"(?P<mac>[0-9a-z:]+) .+\n(?:(?:.+\n\s+|.*)altname \w+)?"
-            r"(?:\s+inet (?P<ip_addr>[\d.]+)\/.*\n)?"
+            r"\d+: (?P<name>\w+): \<.*\>.*link\/(?:ether|infiniband|loopback) "
```
Comment (squirrelsc) [inline lisa/tools/ip.py]: Does it need to update examples?

File: lisa/tools/ip.py [hunk 2]
```diff
@@ -224,16 +231,15 @@ def get_info(self, nic_name: Optional[str] = None) -> List[IpInfo]:
...
             ),
         )
-        entries = find_groups_in_lines(
-            result.stdout, self.__ip_addr_show_regex, single_line=False
-        )
+        raw_list = re.finditer(self.__ip_details_pattern, result.stdout)
         found_nics: List[IpInfo] = []
-        for entry in entries:
+        for ip_raw in raw_list:
+            matched = self._get_matched_dict("".join(ip_raw.group().splitlines()))
```
Comment (squirrelsc) [inline lisa/tools/ip.py]: It doesn't need to split and join. Set the `single_line=False`.

File: lisa/tools/ip.py (hunk 1 above)
Comment (LiliDeng) [reply lisa/tools/ip.py]: updated

File: lisa/tools/ip.py (hunk 2 above)
Comment (LiliDeng) [reply lisa/tools/ip.py]: it doesn't work when string contains `\r\n`

Comment (squirrelsc) [reply lisa/tools/ip.py]: Do you mean, the `single_line==False` works for `\n` only? Can you change it to support both? Like to replace `\r\n` to `\n`.

Comment (LiliDeng) [reply lisa/tools/ip.py]: I'll try.

Comment (LiliDeng) [reply lisa/tools/ip.py]: updated

**Review by squirrelsc — APPROVED** [review]

File: lisa/tools/ip.py (hunk 2 above)
Comment (squirrelsc) [reply lisa/tools/ip.py]: It's not using the utils with `single_line==False`. What's the reason?

Comment (LiliDeng) [reply lisa/tools/ip.py]: when I use find_groups_in_lines single_line = False / True with current pattern, I can't get the value what I want, probably ip_addr, mac and name are not in the same group. previous pattern works well except for below output

```
  3: eth1: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN group default qlen 1000  # noqa: E501
        link/ether 00:15:5d:33:ff:0b brd ff:ff:ff:ff:ff:ff
```

**Pull request merged**